src/Solvers/Greedy.jl

```diff
@@ -172,6 +178,55 @@ function greedy_choose_simple!(queue, remaining)
     return node
 end
 
+
+function greedy_choose_thermal!(queue, remaining, nbranch=8, temperature=1, rel_temperature=true)
+    n = 0
+    choices = Vector{HeapNode}()
+
+    while (!isempty(queue) && n < nbranch)
+        node = pop!(queue)
+        a, b, c = meta(node)
+
+        any(inds -> inds ∉ values(remaining), (a, b)) && continue
+
+        push!(choices, node)
+        n += 1
+    end
+
+    if n == 0
+        return nothing
+    elseif n == 1
+        return choices[1]
+    end
+
+    costs = [choice.cost for choice in choices]
+    cmin = costs[1]
+
+    rel_temperature && (temperature *= max(1, abs(cmin)))
+
+    if isapprox(temperature, 0.)
+        energies = [c == cmin ? 1 : 0 for c in costs]
+    else
+        energies = [exp(-(c - cmin) / temperature) for c in costs]
+    end
```

Comment:
This code is type unstable. Also it can be refactored in a much cleaner way:
```suggestion
    energies = isapprox(temperature, 0.) ? Float32.(costs .== cmin) : exp.(-(costs .- cmin) / temperature)
```